src/test/java/com/conveyal/datatools/manager/jobs/ArbitraryTransformJobTest.java

```diff
@@ -0,0 +1,271 @@
+        feedSource.transformRules.add(transformRules);
+        Persistence.feedSources.replace(feedSource.id, feedSource);
+        // Create target version (note: GTFS folder has no stop_attributes.txt file).
+        targetVersion = createFeedVersion(
+            feedSource,
+            zipFolderFiles("fake-agency-with-only-calendar-dates")
+        );
+        // Check that new version has stop_attributes file
+        ZipFile zip = new ZipFile(targetVersion.retrieveGtfsFile());
+        ZipEntry entry = zip.getEntry(table + ".txt");
+        assertThat(entry, Matchers.notNullValue());
+        // TODO Verify that stop_attributes file matches source file exactly?
+    }
+
+    @Test
+    public void canDeleteTrips() throws IOException {
+        // Add delete trips transformation.
+        List<String> routeIds = new ArrayList<>();
+        // Collect route_id values.
+        routeIds.add("1");
+        // Store the number of trips that run on the route_ids here.
+        int numberOfTripsForRoutes = 1;
+        FeedTransformation transformation = DeleteRecordsTransformation.create(
+            "trips",
+            "route_id",
+            routeIds
+        );
+        FeedTransformRules transformRules = new FeedTransformRules(transformation);
+        feedSource.transformRules.add(transformRules);
+        Persistence.feedSources.replace(feedSource.id, feedSource);
+        // Load feed.
+        sourceVersion = createFeedVersion(
+            feedSource,
+            zipFolderFiles("fake-agency-with-only-calendar")
+        );
+        // Fetch snapshot where modifications were made and create new version from it.
+        Snapshot snapshotWithModifications = feedSource.retrieveSnapshots().iterator().next();
+        CreateFeedVersionFromSnapshotJob newVersionJob = new CreateFeedVersionFromSnapshotJob(feedSource, snapshotWithModifications, user);
+        newVersionJob.run();
+        // Grab the modified version and check that the trips count matches expectation.
+        FeedVersion newVersion = feedSource.retrieveLatest();
+        assertEquals(
+            "trips count for transformed feed should be decreased by the # of records matched by the query",
+            sourceVersion.feedLoadResult.trips.rowCount - numberOfTripsForRoutes,
+            newVersion.feedLoadResult.trips.rowCount
+        );
+    }
+
+    @Test
+    public void canCloneZipFileAndTransform() throws IOException, SQLException {
+        // Generate random UUID for feedId, which gets placed into the csv data.
+        final String feedId = UUID.randomUUID().toString();
+        final String feedInfoContent = generateFeedInfo(feedId);
+        FeedTransformation transformation = ReplaceFileFromStringTransformation.create(feedInfoContent, "feed_info");
+        // Create transform rules for VERSION_CLONE retrieval method.
+        FeedTransformRules transformRules = new FeedTransformRules(transformation, FeedRetrievalMethod.VERSION_CLONE);
```

Comment:
What real-life use case is this trying to model? Why would someone want to do this instead of the transformation done in the `canReplaceFeedInfo` test case?

Reply:
Will address in modifications to how versions are created by transformations.